spec/models/dynamic_field_spec.rb

```diff
@@ -0,0 +1,186 @@
+      its(:sort_order) { is_expected.to be 7 }
+      its(:filter_label) { is_expected.to eql 'Name' }
+      its(:is_facetable) { is_expected.to eql true }
+      its(:controlled_vocabulary) { is_expected.to eql 'name_role' }
+      its(:select_options) { is_expected.to be_nil }
+      its(:is_keyword_searchable) { is_expected.to be false }
+      its(:is_title_searchable) { is_expected.to be false }
+      its(:is_identifier_searchable) { is_expected.to be false }
+      its(:dynamic_field_group) { is_expected.to be_a DynamicFieldGroup }
+    end
+
+    context 'when missing string_key' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, string_key: nil) }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'String key can\'t be blank'
+      end
+    end
+
+    context 'when missing field_type' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, field_type: nil) }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'Field type can\'t be blank'
+      end
+    end
+
+    context 'when field_type invalid' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, field_type: 'not-valid') }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'Field type is not included in the list'
+      end
+    end
+
+    context 'when missing dynamic_field_group' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, dynamic_field_group: nil) }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'Dynamic field group must exist'
+      end
+    end
+
+    context 'when missing created_by' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, created_by: nil) }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'Created by must exist'
+      end
+    end
+
+    context 'when missing updated_by' do
+      let(:dynamic_field) { FactoryBot.build(:dynamic_field, updated_by: nil) }
+
+      it 'does not save' do
+        expect(dynamic_field.save).to be false
+      end
+
+      it 'returns correct error' do
+        dynamic_field.save
+        expect(dynamic_field.errors.full_messages).to include 'Updated by must exist'
```

Comment:
```suggestion
        expect(dynamic_field.errors.full_messages).to include 'Updated by value is required'
```

Reply:
This error message comes from the belongs_to association. Belongs_to associations are required in rails 5, unless specified otherwise. We could add in an additional validation to check for presence. It would be redundant, but would allow us to change the error messaging.

Reply:
I don't like the error redundancy that the user would see in the additional validation case, but I just feel like the phrasing "updated by must exist" is a little odd for a user seeing this in an error message.  My feelings about this aren't too strong though, so I can live with it as is.

Reply:
I get that, I just didn't see a way to easy change that. So I say we leave it as is in order to save time.